src/Foundatio.Aliyun/Storage/AliyunFileStorage.cs

```diff
@@ -89,7 +114,7 @@ public class AliyunFileStorage : IFileStorage {
             }
             if (!stream.CanSeek) {

```

Comment:
Is there any reason we can't skip this whole if block and just pass the stream to PutObjectRequest?

Reply:
In the OssClient, when saving files with PutObject method, it will call the `stream.Length` property. But when the stream cannot be seek, there will be `NotSupportedException` thrown, e.g. HttpResponseStream, NetWorkStream etc. Please refer to the unit test https://github.com/FoundatioFx/Foundatio/blob/c092b26d42f1bd3fc7a09aaca48054b92c92a5fa/test/Foundatio.TestHarness/Storage/FileStorageTestsBase.cs#L165
So we have to copy the stream to memory to support stream that cannot be seek, since we cannot change the implementation of Aliyun OSS SDK. Although there will be memory allocation. Do you have any better suggestion?